### ISSUE TITLE
createCatalogItemFromUrl.register is not the source of truth

### ISSUE THREAD

**Issue by RacingTadpole** [TerriaJS/terriajs#1718]
`AddData.jsx` hardcodes all the catalog items to check for. So `registerCatalogMembers`'s calls to `createCatalogItemFromUrl` are irrelevant.

We should either remove these calls, or use them.  And write some tests while we're at it.

From #1717.

**Comment by kring** [comment]
That's not true, is it?  `createCatalogItemFromFileOrUrl` is used in the `auto` case, right?

**Comment by kring** [comment]
It looks like it would be pretty easy to replace all the `loadXxxx` functions with something more generic though.

**Comment by RacingTadpole** [comment]
Oh yes, I missed that. So `AddData`'s `handleUrl` tries wms, wfs, wmts, mapserver, mapserverlayer, and then finishes by calling `loadFile` which throws it back to `createCatalogItemFromFileOrUrl` and thus (if type is auto) to `createCatalogItemFromUrl`.

I could rephrase this as: `createCatalogItemFromUrl.register` is misleading, because it's not the whole story - even if you leave `wms` off, it will still try it (in fact before those you do register).

**Comment by RacingTadpole** [comment]
Do we need to test those 5 special types before trying the ones specified by `createCatalogItemFromUrl.register`?  The only rationale I can see is it uses some smarts to decide which of WMS and WFS should go first.
I've just tested a version on my computer without this and find it works ok for ArcGisMapServer and WMS, at least.

**Comment by kring** [comment]
Maybe not.  There has been some evolution of the code in this area recently.  The key test to do is to add a `/wfs` URL that also can talk WMS (e.g. any geoserver pretty much) and make sure you get WFS not WMS.

**Comment by RacingTadpole** [comment]
I tried my version with https://www....gov.au/geoserver/wfs (which I believe can also talk WMS) and it came up as WFS. For comparison, nationalmap.gov.au failed to load that URL.

**Comment by stevage** [comment]
Yeah, I'm sorry, I broke some things when I refactored this code a while ago - I missed some of the intended behaviour.
